components/hab/src/cli/hab/util.rs

```diff
@@ -18,24 +25,84 @@ use structopt::StructOpt;
 use url::Url;
 
 #[derive(ConfigOpt, StructOpt)]
+#[configopt(derive(Serialize))]
 #[structopt(no_version)]
-#[allow(dead_code)]
 pub struct AuthToken {
-    /// Authentication token for Builder
+    /// Authentication token for Builder.
+    // TODO (JM): This should probably use `env`
     #[structopt(name = "AUTH_TOKEN", short = "z", long = "auth")]
-    auth_token: Option<String>,
+    pub value: Option<String>,
 }
 
 #[derive(ConfigOpt, StructOpt, Deserialize)]
 #[configopt(derive(Serialize))]
 #[structopt(no_version)]
-#[allow(dead_code)]
 pub struct BldrUrl {
-    /// Specify an alternate Builder endpoint. If not specified, the value will be taken from
-    /// the HAB_BLDR_URL environment variable if defined. (default: https://bldr.habitat.sh)
+    /// Specify an alternate Builder endpoint. If not specified, the value will be
+    /// taken from the HAB_BLDR_URL environment variable if defined. (default: https://bldr.habitat.sh)
     // TODO (DM): This should probably use `env` and `default_value`
     #[structopt(name = "BLDR_URL", short = "u", long = "url")]
-    bldr_url: Option<Url>,
+    pub value: Option<Url>,
+}
+
+#[derive(ConfigOpt, StructOpt, Deserialize, Serialize)]
+#[structopt(no_version)]
+#[configopt(derive(Serialize))]
+pub struct BldrOrigin {
+    /// The Builder origin name to target
+    #[structopt(name = "ORIGIN", short = "o", long = "origin")]
+    pub inner: Origin,
+}
+
+pub fn bldr_url_from_env_load_or_default() -> String {
```

Comment:
This could be private.

Reply:
Changed to private in https://github.com/habitat-sh/habitat/pull/7841/commits/78f9d5e6a2b76e641bb6f2a2971c62744b5e762b